iceoryx_examples/complexdata/iox_subscriber_complexdata.cpp

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "topic_data.hpp"
+
+#include "iceoryx_posh/popo/subscriber.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+#include "iceoryx_utils/cxx/string.hpp"
+#include "iceoryx_utils/posix_wrapper/signal_handler.hpp"
+
+bool killswitch = false;
+constexpr char APP_NAME[] = "iox-cpp-subscriber-complexdata";
+
+static void sigHandler(int f_sig [[gnu::unused]])
+{
+    // caught SIGINT or SIGTERM, now exit gracefully
+    killswitch = true;
+}
+
+int main()
+{
+    // register sigHandler
+    auto signalIntGuard = iox::posix::registerSignalHandler(iox::posix::Signal::INT, sigHandler);
+    auto signalTermGuard = iox::posix::registerSignalHandler(iox::posix::Signal::TERM, sigHandler);
+
+    // initialize runtime
+    iox::runtime::PoshRuntime::initRuntime(APP_NAME);
+
+    // initialize subscriber
+    iox::popo::Subscriber<ComplexDataType> subscriber({"Radar", "FrontLeft", "Object"});
+
+    // run until interrupted by Ctrl-C
+    while (!killswitch)
+    {
+        subscriber.take()
+            .and_then([](auto& sample) {
+                std::stringstream s;
+                s << APP_NAME << " got values:";
+
+                s << std::endl << "stringForwardList: ";
+                for (auto i = sample->stringForwardList.begin(); i != sample->stringForwardList.end(); ++i)
+                {
+                    s << *i << " ";
+                }
```

Comment:
just a small hint, if you omit the white-space after `stringForwardList: ` and make `s << " " << *i;` there is no superfluous white-space at the end of the line :)